x-pack/legacy/plugins/lens/public/indexpattern_plugin/state_helpers.ts

```diff
@@ -158,3 +160,26 @@ export function getColumnOrder(columns: Record<string, IndexPatternColumn>): str
     .map(([id]) => id)
     .concat(metrics.map(([id]) => id));
 }
+
+export function isLayerTransferable(layer: IndexPatternLayer, newIndexPattern: IndexPattern) {
+  return Object.values(layer.columns).every(column =>
+    isColumnTransferable(column, newIndexPattern)
+  );
+}
+
+export function updateLayerIndexPattern(
+  layer: IndexPatternLayer,
+  newIndexPattern: IndexPattern
+): IndexPatternLayer {
+  const newColumns: IndexPatternLayer['columns'] = _.pick(layer.columns, column =>
+    isColumnTransferable(column, newIndexPattern)
+  );
```

Comment:
This logic isn't going as far as I was expecting to reset any defaults that might need to be changed. For example, does the operation definition need to change when switching from rollup -> regular index?